### ISSUE TITLE
The first example on the README is broken as given

### ISSUE THREAD

**Issue by mtravers** [metasoarous/oz#92]
It produces a blank screen and an impenetrable javascript error. I did figure out the problem though, it was the lack of types in line-plot. This version works:

```
(def line-plot
  {:data {:values (play-data "monkey" "slipper" "broom")}
   :encoding {:x {:field "time" :type "quantitative"}
              :y {:field "quantity" :type "quantitative"}
              :color {:field "item" :type "nominal"}}
   : mark "line"})
```

Not the best first experience!

**Comment by sdaro** [comment]
I can confirm this issue and the solution works for me.
Do you like to create a pull request to make it easier for the maintainer?
Or should I?

**Comment by mtravers** [comment]
I went ahead and made a PR: https://github.com/metasoarous/oz/pull/93

**Comment by metasoarous** [comment]
Hi @mtravers. Thanks for the report and the fix (and you @sdaro for verifying)! I fully agree that this is not the bets first experience with Oz.

Do you know if all of the types are necessary? If just the binned type is required, I'd be tempted to only explicitly declare the type there. I'm generally lazy with the types when I can be, to save typing (like in the physical sense). Maybe I should bite the bullet and stop doing this to avoid issues like this in the future though? Open to thoughts here.

Thanks again!

**Comment by mtravers** [comment]
I assumed that because both fields were quantitative they were treated similarly and would both need types. Looks like I was right.

If there's some way to make the types optional I'd be all for it, but the real problem here was that the failure was utterly incomprehensible. I only figured it out by rebuilding Oz without optimization. So, better error handing would be good.

**Comment by metasoarous** [comment]
@mtravers Thanks for the feedback!

You're absolutely right; There's definitely a need for improved error handling (please see #62).

**Comment by metasoarous** [comment]
I've merged #93, and so am going to close this now. Please follow or chime in on #62 for updates or further thoughts on error handling.

Final question: Were you able to get through the rest of the README content without any issues?

Thanks again!
